app/controllers/WhatIsNextController.scala

```diff
@@ -67,22 +70,54 @@ class WhatIsNextController @Inject()(
 
         value => {
           for {
+            hasAnswerChanged <- Future.fromTry(Success(!request.userAnswers.get(WhatIsNextPage).contains(value)))
             updatedAnswers <- Future.fromTry(request.userAnswers.set(WhatIsNextPage, value))
             _ <- playbackRepository.set(updatedAnswers)
+            result <- updateMigrationStatusAndRedirect(value, hasAnswerChanged)
           } yield {
-            val call = value match {
-              case DeclareTheTrustIsUpToDate => redirectToDeclaration
-              case MakeChanges => redirectToFirstUpdateQuestion
-              case CloseTrust if request.userAnswers.isTrustTaxable => controllers.close.taxable.routes.DateLastAssetSharedOutYesNoController.onPageLoad()
-              case CloseTrust => controllers.close.nontaxable.routes.DateClosedController.onPageLoad()
-              case NoLongerTaxable => controllers.routes.NoTaxLiabilityInfoController.onPageLoad()
-              case NeedsToPayTax => controllers.routes.FeatureNotAvailableController.onPageLoad()
-              case GeneratePdf => controllers.routes.ObligedEntityPdfController.getPdf(request.userAnswers.identifier)
-            }
-
-            Redirect(call)
+            result
           }
         }
       )
   }
+
+  private def updateMigrationStatusAndRedirect(newAnswer: WhatIsNext, hasAnswerChanged: Boolean)
+                                              (implicit request: DataRequest[AnyContent]): Future[Result] = {
+
+    def redirect(implicit request: DataRequest[AnyContent]): Result = Redirect {
+      newAnswer match {
+        case DeclareTheTrustIsUpToDate =>
+          redirectToDeclaration
+        case MakeChanges =>
+          redirectToFirstUpdateQuestion
+        case CloseTrust if request.userAnswers.isTrustTaxable =>
+          controllers.close.taxable.routes.DateLastAssetSharedOutYesNoController.onPageLoad()
+        case CloseTrust =>
+          controllers.close.nontaxable.routes.DateClosedController.onPageLoad()
+        case NoLongerTaxable =>
+          controllers.routes.NoTaxLiabilityInfoController.onPageLoad()
+        case NeedsToPayTax =>
+          controllers.routes.FeatureNotAvailableController.onPageLoad()
+        case GeneratePdf =>
+          controllers.routes.ObligedEntityPdfController.getPdf(request.userAnswers.identifier)
+      }
+    }
+
+    if (newAnswer != GeneratePdf) {
+      for {
+        _ <- {
+          if (hasAnswerChanged) {
+            trustConnector.removeTransforms(request.userAnswers.identifier)
```

Comment:
For this one you could do `removeTransforms().map(_ => ())` to return unit and do the same on the else block of `Future.successful(())` rather than a OK

Reply:
Oh that's much better. Will amend that now